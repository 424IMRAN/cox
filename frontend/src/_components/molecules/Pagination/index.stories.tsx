import React from 'react'
import { Meta } from '@storybook/react'
import { ThemeProvider } from '@material-ui/core/styles'
import TablePaginationDemo from './index'
import theme from '../../../core-utils/theme'
import Pagination from './index'

export default {
  title: 'molecules/Pagination',
  component: TablePaginationDemo,
  argTypes: {
    totalItems: {
      control: { type: 'number' },
    },
    itemsPerPage: {
      control: { type: 'number' },
    },
  },
} as Meta

const Default = () => <Pagination rowsPerPage={1} page={1} totalRows={2} />

export const Primary = Default.bind({})
