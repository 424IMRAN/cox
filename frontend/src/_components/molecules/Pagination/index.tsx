import React from 'react'
import { TablePagination } from '@material-ui/core'
import { makeStyles } from '@material-ui/styles'
import clsx from 'clsx'
import { COLORS } from '../../../core-utils/theme'

const useStyles = makeStyles(() => ({
  paginationStyle: {
    backgroundColor: COLORS.TRANSPARENT_GREY_500,
  },
}))

type PaginationProps = {
  page: number
  setPage?: (currentPage: number) => void
  rowsPerPage: number
  setRowsPerPage?: (rows: number) => void
  totalRows: number
  paginationStyle?: string
  [restProps: string]: any
}

const Pagination = ({
  page,
  setPage,
  rowsPerPage,
  setRowsPerPage,
  totalRows,
  paginationStyle,
  ...restProps
}: PaginationProps) => {
  const classes = useStyles()

  const handleChangePage = (
    event: React.MouseEvent<HTMLButtonElement> | null,
    newPage: number
  ) => {
    setPage?.(newPage)
  }

  const handleChangeRowsPerPage = (
    event: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>
  ) => {
    setRowsPerPage?.(parseInt(event.target.value, 10))
    setPage?.(0)
  }

  return (
    <TablePagination
      className={clsx(classes.paginationStyle, paginationStyle)}
      component="div"
      count={totalRows}
      page={page}
      onPageChange={handleChangePage}
      rowsPerPage={rowsPerPage}
      onRowsPerPageChange={handleChangeRowsPerPage}
      {...restProps}
    />
  )
}

export default Pagination

Pagination.defaultProps = {
  setPage: null,
  setRowsPerPage: () => null,
  paginationStyle: null,
}
