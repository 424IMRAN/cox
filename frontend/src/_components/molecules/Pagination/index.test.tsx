import { render } from '@testing-library/react'
import { TablePagination } from '@material-ui/core'
import { mount } from 'enzyme'
import Pagination from './index'

describe('Pagination Testing', () => {
  it('Matches Snapshot', () => {
    const { container } = render(
      <Pagination rowsPerPage={1} page={1} totalRows={1} />
    )
    expect(container).toMatchSnapshot()
  })

  it('Allows us to set props', () => {
    const wrapper = mount(<Pagination rowsPerPage={1} page={1} totalRows={1} />)
    expect(wrapper.props().rowsPerPage).toEqual(1)
    expect(wrapper.props().page).toEqual(1)
    expect(wrapper.props().totalRows).toEqual(1)
    wrapper.setProps({ page: 2 })
    wrapper.setProps({ rowsPerPage: 2 })
    expect(wrapper.props().page).toEqual(2)
    expect(wrapper.props().rowsPerPage).toEqual(2)
  })

  it('onChange works', () => {
    const mockHandleChange = jest.fn()
    const component = mount(
      <Pagination
        rowsPerPage={1}
        page={1}
        totalRows={2}
        setPage={mockHandleChange}
      />
    )

    component
      ?.find(TablePagination)
      ?.props()
      ?.onPageChange?.({} as any, 2)
    expect(mockHandleChange).toHaveBeenCalledWith(2)
  })
})
