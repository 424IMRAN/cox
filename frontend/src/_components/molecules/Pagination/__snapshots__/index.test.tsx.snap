// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination Testing Matches Snapshot 1`] = `
<div>
  <div
    class="MuiTablePagination-root makeStyles-paginationStyle-1"
  >
    <div
      class="MuiToolbar-root MuiToolbar-regular MuiTablePagination-toolbar MuiToolbar-gutters"
    >
      <div
        class="MuiTablePagination-spacer"
      />
      <p
        class="MuiTypography-root MuiTablePagination-caption MuiTypography-body2 MuiTypography-colorInherit"
        id="mui-24324"
      >
        Rows per page:
      </p>
      <div
        class="MuiInputBase-root MuiTablePagination-input MuiTablePagination-selectRoot"
      >
        <div
          aria-haspopup="listbox"
          aria-labelledby="mui-24324 mui-49962"
          class="MuiSelect-root MuiSelect-select MuiTablePagination-select MuiSelect-selectMenu MuiInputBase-input"
          id="mui-49962"
          role="button"
          tabindex="0"
        >
          <span>
            â€‹
          </span>
        </div>
        <input
          aria-hidden="true"
          class="MuiSelect-nativeInput"
          tabindex="-1"
          value="1"
        />
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSelect-icon MuiTablePagination-selectIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M7 10l5 5 5-5z"
          />
        </svg>
      </div>
      <p
        class="MuiTypography-root MuiTablePagination-caption MuiTypography-body2 MuiTypography-colorInherit"
      >
        2-1 of 1
      </p>
      <div
        class="MuiTablePagination-actions"
      >
        <button
          aria-label="Previous page"
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit"
          tabindex="0"
          title="Previous page"
          type="button"
        >
          <span
            class="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M15.41 16.09l-4.58-4.59 4.58-4.59L14 5.5l-6 6 6 6z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
        <button
          aria-label="Next page"
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit Mui-disabled Mui-disabled"
          disabled=""
          tabindex="-1"
          title="Next page"
          type="button"
        >
          <span
            class="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M8.59 16.34l4.58-4.59-4.58-4.59L10 5.75l6 6-6 6z"
              />
            </svg>
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;
